src/DistributedMetadata/PlacementStrategy.cpp

```diff
@@ -16,6 +18,18 @@ std::vector<NodeMetricsPtr> DiskStrategy::qualifiedNodes(const NodeMetricsContai
 
     for (const auto & [node, metrics] : nodes_metrics)
     {
+        /// If the role of the node is "ingest" or "search", skip it.
+        std::vector<String> roles;
+        boost::split(roles, metrics->node.roles, boost::is_any_of(","));
```

Comment:
Not sure if it's a good idea that we use a string to indicate node roles, maybe we can use unsigned integer to do the same things, then we can use bit operation. It's faster than what we did here

Reply:
This is config from config.xml, we don't need convert to bit or integer. It is highly readable, loggable and no perf issue at all.